crates/transport-nats/src/lib.rs

```diff
@@ -450,18 +460,34 @@ impl AsyncWrite for SubjectWriter {
 
     #[instrument(level = "trace", skip_all, ret, fields(subject = self.tx.as_str()))]
     fn poll_shutdown(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<std::io::Result<()>> {
-        trace!("writing empty buffer to shut down stream");
+        trace!("writing stream shutdown message");
         ready!(self.as_mut().poll_write(cx, &[]))?;
+        self.shutdown = true;
         Poll::Ready(Ok(()))
     }
 }
 
+impl Drop for SubjectWriter {
+    fn drop(&mut self) {
+        if !self.shutdown {
+            let nats = self.nats.clone();
+            let subject = mem::replace(&mut self.tx, Subject::from_static(""));
+            tokio::spawn(async move {
```

Comment:
I've heard conflicting reports about whether or not you should be using `tokio::spawn` in `Drop`. The argument against it is that Drop could panic if it is not inside of a runtime. The argument for is that "if you know everything is running in tokio" then it is fine. 

To be safe here, should this use `Runtime::try_current()` to try and get and handle, and then spawn?

Reply:
I've started off using `try_current` and replaced it with this approach for simplicity.
It's impossible to construct this type outside a Tokio runtime, but I suppose one could manually `thread::spawn` and move this stream there.
Again, there would be nothing they could do with this stream outside a Tokio runtime other than drop it.

So I guess it is theoretically possible to move this into a separate thread just for the purpose of dropping it - that sounds like a bug, but we probably should not panic in that case.